src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,180 @@
+            
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName(
+                            WellKnownTypes.SystemNetSecurityRemoteCertificateValidationCallback);
+                    if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            var delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol.Equals(delegateCreationOperation.Type))
+                            {
+                                var alwaysReturnTrue = false;
+                                var kindOfTargetFunction = delegateCreationOperation.Target.Kind;
+
+                                switch (kindOfTargetFunction)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        alwaysReturnTrue = AlwaysReturnTrue(delegateCreationOperation.Target.Descendants());
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        var methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        var methodSymbol = methodReferenceOperation.Method;
+                                        var blockOperation = methodSymbol.GetTopmostOperationBlock(compilationStartAnalysisContext.Compilation);
+                                        // TODO(LINCHE): This is an issue tracked by #2009. We filter extraneous based on IsImplicit.
+                                        var targetOperations = FilterImplicitOperations(ImmutableArray.ToImmutableArray(blockOperation.Descendants()));
+                                        alwaysReturnTrue = AlwaysReturnTrue(targetOperations);
+                                        break;
+                                }
+
+                                if (alwaysReturnTrue)
+                                {
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        delegateCreationOperation.CreateDiagnostic(
+                                            Rule,
+                                            kindOfTargetFunction.ToString()));
```

Comment:
I don't think kindOfTargetFunction ("AnonymousMethod" / "MethodReference") will be useful information for developers.  It'd be better to have the name of the method being referenced (for delegates using non-local or local methods) or "anonymous method" for the anonymous case.   Or not have a message parameter at all.